ui/src/pages/ShoeMap.tsx

```diff
@@ -101,9 +101,31 @@ export const ShoeMap: React.FC = () => {
           <TileLayer
             url={`https://api.mapbox.com/styles/v1/hanlinc27/ckhjy5wat2dvz1aplv4tkaghb/tiles/{z}/{x}/{y}?access_token=${process.env.REACT_APP_MAPBOX_ACCESS_TOKEN}`}
           />
-          {stories && !error && (
-            <PinCluster stories={stories} openDrawer={handleOpenDrawer} />
-          )}
+          {(() => {
+
+            if(stories && filteredStories.length != 0){
+              var filteredStoryArr= new Array<Story>();
```

Comment:
1) You can find the intersection between two arrays with `.filter` and `.includes`. So something like:
```
const filteredStoryArr = stories.filter(value => filterStories.includes(value.author_country));
```
^ This is not super optimum, but works well for our case. If you are wondering about performance, we can also use es6 sets (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) or memoize with a hash map.

2) We shouldn't need to call a anon function with `(() => {...})()`. Instead we can compute this before the return value. So something like:

```js
const { data, error } = useSWR<Story[]>("/api/stories");
const [filteredCountries, setFilteredCountries] = useState([]);

...

// If no filtered countries just make stories equal to data
// otherwise filter the stories based on the intersection of the two arrays.
const stories = filteredCountries.length === 0 ? data : data.filter(story => filterStories.includes(story.author_country));

...

return (
  ...
  ...
  
  <PinCluster stories={stories} openDrawer={handleOpenDrawer} />
  ...
  ...
```